src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -534,6 +534,8 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
 
     @JvmStatic fun companionObjectBuilder() = Builder(Kind.COMPANION, null, null)
 
+    @JvmStatic fun companionObjectBuilder(name: String) = Builder(Kind.COMPANION, name, null)
```

Comment:
Should this be nullable and have a default of null (and `@JvmOverloads`). It's not very idiomatic Kotlin to have manual overloads like this.